test/e2e/azure_test.go

```diff
@@ -118,6 +118,12 @@ var _ = Describe("Workload cluster creation", func() {
 			_ = bootstrapClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: clusterName, Namespace: namespace.Name}, result.Cluster)
 		}
 
+		if CurrentGinkgoTestDescription().Failed {
```

Comment:
what do you think about making the same changes in the other `AfterEach` functions (conformance_test.go and capi_test.go)? Or even better, extracting the common logic to a helper function we can reuse everywhere

Reply:
I think that's a great idea! I'll work on making this a helper function and adding it to the other functions.

Reply:
Made this change in my latest commit, let me know if it looks good!